h3py/h3core.pyx

```diff
@@ -0,0 +1,274 @@
+
+    return mv
+
+
+
+cpdef H3int[:] children(H3int h, res=None):
+    u.check_addr(h)
+
+    if res is None:
+        res = resolution(h) + 1
+    # todo: actually, do we want to raise an error if there are no children, or just return an empty set?
+    u.check_res(res)
+
+    n = h3c.maxH3ToChildrenSize(h, res)
+
+    ptr = create_ptr(n)
+    h3c.h3ToChildren(h, res, ptr)
+    mv = create_mv(ptr, n)
+
+    return mv
+
+
+
+cpdef H3int[:] compact(const H3int[:] hu):
+    for h in hu:
+        u.check_addr(h)
+
+    ptr = create_ptr(len(hu))
+    flag = h3c.compact(&hu[0], ptr, len(hu))
+    mv = create_mv(ptr, len(hu))
+
+    if flag != 0:
+        raise H3ValueError('Could not compact set of hexagons!')
+
+    return mv
+
+# todo: https://stackoverflow.com/questions/50684977/cython-exception-type-for-a-function-returning-a-typed-memoryview
+# apparently, memoryviews are python objects, so we don't need to do the except clause
+cpdef H3int[:] uncompact(const H3int[:] hc, int res):
+    for h in hc:
+        u.check_addr(h)
+
+    N = h3c.maxUncompactSize(&hc[0], len(hc), res)
+
+    ptr = create_ptr(N)
+    flag = h3c.uncompact(
+        &hc[0], len(hc),
+           ptr, N,
+        res
+    )
+    mv = create_mv(ptr, N)
+
+    if flag != 0:
+        raise H3ValueError('Could not uncompact set of hexagons!')
+
+    return mv
+
+
+
+# weird return type here
+cpdef H3int num_hexagons(int resolution) except -1:
+    u.check_res(resolution)
+
+    return h3c.numHexagons(resolution)
+
+
+cpdef double hex_area(int resolution, unit='km') except -1:
+    u.check_res(resolution)
+
+    area = h3c.hexAreaKm2(resolution)
+
+    # todo: multiple units
+    convert = {
+        'km': 1.0,
+    }
+    area *= convert[unit]
+
+    return area
+
+cpdef double edge_length(int resolution, unit='km') except -1:
+    u.check_res(resolution)
+
+    length = h3c.edgeLengthKm(resolution)
+
+    # todo: multiple units
+    convert = {
+        'km': 1.0,
+    }
+    length *= convert[unit]
+
+    return length
+
+
+
+
+cpdef bool are_neighbors(H3int h1, H3int h2):
+    u.check_addr(h1)
+    u.check_addr(h2)
+
+    return h3c.h3IndexesAreNeighbors(h1, h2) == 1
+
+
+cpdef H3int edge(H3int origin, H3int destination) except 1:
+    u.check_addr(origin)
+    u.check_addr(destination)
+
+    if h3c.h3IndexesAreNeighbors(origin, destination) != 1:
+        raise H3ValueError('Hexes are not neighbors: {} and {}'.format(origin, destination))
+
+    return h3c.getH3UnidirectionalEdge(origin, destination)
+
+
+cpdef bool is_edge(H3int e):
+    u.check_edge(e)
+
+    return h3c.h3UnidirectionalEdgeIsValid(e) == 1
+
+cpdef H3int edge_origin(H3int e) except 1:
+    u.check_edge(e)
+
+    return h3c.getOriginH3IndexFromUnidirectionalEdge(e)
+
+cpdef H3int edge_destination(H3int e) except 1:
+    u.check_edge(e)
+
+    return h3c.getDestinationH3IndexFromUnidirectionalEdge(e)
+
+cpdef (H3int, H3int) edge_hexes(H3int e) except *:
```

Comment:
`edge_indexes` is likely a better term here, as with above, when there's a possibility of pentagons we should prefer the term `index` for the representation of the cell.

Reply:
how about `edge_cells`?

Reply:
I've updated the naming to match the current python implementation.